juriscraper/OpinionSite.py

```diff
@@ -108,6 +109,9 @@ def _get_precedential_statuses(self):
     def _get_summaries(self):
         return None
 
+    def _get_child_courts(self):
+        return None
+
     def extract_from_text(self, scraped_text):
```

Comment:
What do you think about this?
``` 
   @staticmethod
    def match(scraped_text, pattern):
        m = re.findall(pattern, scraped_text)
        return next((text for group in m for r in group if (text := r)), "")

    def extract_from_text(self, scraped_text: str) -> Dict[str, Any]:
        """"""
        # Find Author Str
        pattern = r"<td[^>]*>(.*?),\s?[JS]\.</td>|Judge:\s?(.+)|([\w .,]+), J\.\s+"
        author_str = self.match(scraped_text, pattern)

        # Find Docket Number
        pattern = r"</table><br><br\s?/?>\s?(.*)\r\n|Docket Number:\s?(.+)"
        docket_number = harmonize(self.match(scraped_text, pattern))

        # Find Misc Citation
        pattern = r"\[(?P<volume>\d+) (?P<reporter>Misc [23]d) (?P<page>.+)\]"
        cite_match = re.search(pattern, scraped_text)

        return {
            "Docket": {"docket_number": docket_number},
            "Opinion": {"author_str": normalize_judge_string(author_str)[0]},
            "Citation": cite_match.groupdict("") if cite_match else {},
        }
```

Reply:
I took a look at it - and I think we can accomplish all of this with just three simple regex patterns.  I think ... it might keep the complexity of maintaining it easier?

Reply:
Also - I obviously put it in the wrong file - so apologies of the confusion, but I don't want to move it.